src/pymortests/strategies.py

```diff
@@ -464,11 +464,8 @@ def base_vector_arrays(draw, count=1, dtype=None, max_dim=100):
     length = space.dim
 
     # this lets hypothesis control np's random state too
-    random = draw(hyst.random_module())
-    # scipy performs this check although technically numpy accepts a different range
-    assume(0 <= random.seed < 2**32 - 1)
-    random_correlation = random_correlation_gen(random.seed)
-    rng = np.random.default_rng(0)
+    random = draw(hyst.randoms())
+    rng = np.random.default_rng(random.randint(0, 2**32-1))
```

Comment:
Can you say what's going on here?

Reply:
If we both don't understand the code, mybe we should not touch it unless we better understand it. Maybe open a separate issue?

Reply:
I also don't understand the current code in `main`, so both are equally bad for me in that sense...
What I noticed is that, after removing `np.random.seed(42)` in [`reset_rng`](https://github.com/pymor/pymor/blob/2360b7ca36d5333498cbcc741b300bbe6a8621d6/conftest.py#L61-L64), some tests started failing on CI when they worked fine locally, and after replacing `hyst.random_module` with `hyst.randoms` both CI and running locally worked fine.

Reply:
That is strange. I'd suggest to also move this into a different issue/PR to be able to look at this more carefully. (And also leave the `np.random.seed(42)` in for this PR.)